### DISCUSSION TITLE
Duplicate Packet Handling

### DISCUSSION THREAD

**Discussion by fuzzyupdates** [Security-Onion-Solutions/securityonion#8098] [Unsupported Versions] [answered]
TL;DR:
How does Security Onion, and its packet handling components like AF-PACKET, Stenographer, Surricata, etc., handle actual duplicate packets? I've searched online for different combinations of those tools and the issue of duplicate packet handling and have come up empty.

Context:
Very soon, we will be deploying our distributed environment with a number of forward nodes in different locations that will be sniffing traffic off SPAN ports at their respective locations. All the nodes will be registered to a manager in a centralized location.

For a basic architecture layout, all layer 3 routing interfaces for our different subnets (clients VLAN 10, servers VLAN 20, etc. for example) will be sub-interfaces on firewall hardware. The physical interface will connect to a switch that will server as the SPAN source and down-links to the clients and servers.

As a result, all local client/server communications will traverse the same physical interface as many as four times depending on what it is. There will also be traffic on the physical link we don't want to be sniffing out (other VLANs, etc.).

The result is we'll be performing SPANing of the specific VLANs in receive mode only (`monitor sessions 1 source vlan 10 , 20 rx`). This reduces the maximum sniffed packets from four to two, but obviously still results in duplication hence my question.

If anyone has ideas on how to eliminate the duplication that would be great. But given our architecture I don't think there is a practical way to do it. _So given that there will be truly duplicate packets ingested into SO: how will SO handle them? I.e. will there be duplicate log entries, events, alerts, etc.?_

Thanks!

**Comment by dougburks** [comment, marked as answer]
I suspect you would see duplicate logs. You may be able to filter some traffic using BPF:
https://docs.securityonion.net/en/2.3/bpf.html

**Reply by fuzzyupdates** [reply]
Yeah, I figured that would be the case. We'll dig into the BPF a little deeper to see if we can clean things up, but I have a feeling we're just going to have to live with some duplication.

Thanks, Doug.
